src/lib/ngsi/ContextAttribute.cpp

```diff
@@ -136,18 +172,80 @@ ContextAttribute::ContextAttribute
 */
 ContextAttribute::ContextAttribute
 (
+  const std::string&  _name,
+  const std::string&  _type,
+  double              _value,
+  bool                _found
+)
+{
+  LM_T(LmtClone, ("Creating a number ContextAttribute '%s':'%s':'%d', setting its compound to NULL",
+                  _name.c_str(),
+                  _type.c_str(),
+                  _value));
+
+  name                  = _name;
+  type                  = _type;
+  numberValue           = _value;
+  valueType             = ValueTypeNumber;
+  compoundValueP        = NULL;
+  found                 = _found;
+
+  providingApplication.set("");
+  providingApplication.setFormat(NOFORMAT);
+}
+
+
+
+/* ****************************************************************************
+*
+* ContextAttribute::ContextAttribute -
+*/
+ContextAttribute::ContextAttribute
+(
+  const std::string&  _name,
+  const std::string&  _type,
+  bool                _value,
+  bool                _found
+)
+{
+  LM_T(LmtClone, ("Creating a boolean ContextAttribute '%s':'%s':'%s', setting its compound to NULL",
+                  _name.c_str(),
+                  _type.c_str(),
+                  _value ? "true" : "false"));
+
+  name                  = _name;
+  type                  = _type;
+  boolValue             = _value;
+  valueType             = ValueTypeBoolean;
+  compoundValueP        = NULL;
+  found                 = _found;
+
+  providingApplication.set("");
+  providingApplication.setFormat(NOFORMAT);
+}
+
+
+
+
+/* ****************************************************************************
+*
+* ContextAttribute::ContextAttribute -
+*/
+ContextAttribute::ContextAttribute
+(
   const std::string&         _name,
   const std::string&         _type,
   orion::CompoundValueNode*  _compoundValueP
 )
 {
-  LM_T(LmtClone, ("Creating a ContextAttribute, maintaing a pointer to compound value (at %p)", _compoundValueP));
+  LM_T(LmtClone, ("Creating a ContextAttribute, maintaining a pointer to compound value (at %p)", _compoundValueP));
 
   name                  = _name;
   type                  = _type;
   compoundValueP        = _compoundValueP->clone();
   typeFromXmlAttribute  = "";
   found                 = false;
+  valueType             = ValueTypeCompoundObject;  // FIXME P6: Couls be ValueTypeCompoundVector ...
```

Comment:
Couls -> Could.

However, perhaps we can get the info on Object/Vector from _compoundValueP (and remove the FIXME) ?